Userland/Libraries/LibWeb/Painting/BorderPainting.cpp

```diff
@@ -11,30 +11,35 @@
+BorderRadiiData normalized_border_radii_data(Layout::Node const& node, Gfx::FloatRect const& rect, CSS::BorderRadiusData top_left_radius, CSS::BorderRadiusData top_right_radius, CSS::BorderRadiusData bottom_right_radius, CSS::BorderRadiusData bottom_left_radius, RelativeToWidthOnly relative_to_width_only)
 {
-    auto width_length = CSS::Length::make_px(rect.width());
-    auto height_length = CSS::Length::make_px(rect.height());
-
-    BorderRadiusData bottom_left_radius_px {
-        bottom_left_radius.horizontal_radius.resolved(node, width_length).to_px(node),
-        bottom_left_radius.vertical_radius.resolved(node, height_length).to_px(node)
-    };
-
-    BorderRadiusData bottom_right_radius_px {
-        bottom_right_radius.horizontal_radius.resolved(node, width_length).to_px(node),
-        bottom_right_radius.vertical_radius.resolved(node, height_length).to_px(node)
-    };
+    BorderRadiusData bottom_left_radius_px {};
+    BorderRadiusData bottom_right_radius_px {};
+    BorderRadiusData top_left_radius_px {};
+    BorderRadiusData top_right_radius_px {};
 
-    BorderRadiusData top_left_radius_px {
-        top_left_radius.horizontal_radius.resolved(node, width_length).to_px(node),
-        top_left_radius.vertical_radius.resolved(node, height_length).to_px(node)
-    };
-
-    BorderRadiusData top_right_radius_px {
-        top_right_radius.horizontal_radius.resolved(node, width_length).to_px(node),
-        top_right_radius.vertical_radius.resolved(node, height_length).to_px(node)
-    };
+    auto width_length = CSS::Length::make_px(rect.width());
+    bottom_left_radius_px.horizontal_radius = bottom_left_radius.horizontal_radius.resolved(node, width_length).to_px(node);
+    bottom_right_radius_px.horizontal_radius = bottom_right_radius.horizontal_radius.resolved(node, width_length).to_px(node);
+    top_left_radius_px.horizontal_radius = top_left_radius.horizontal_radius.resolved(node, width_length).to_px(node);
+    top_right_radius_px.horizontal_radius = top_right_radius.horizontal_radius.resolved(node, width_length).to_px(node);
+
+    if (relative_to_width_only == RelativeToWidthOnly::No) {
+        // Normal correct rendering:
+        auto height_length = CSS::Length::make_px(rect.height());
+        bottom_left_radius_px.vertical_radius = bottom_left_radius.vertical_radius.resolved(node, height_length).to_px(node);
+        bottom_right_radius_px.vertical_radius = bottom_right_radius.vertical_radius.resolved(node, height_length).to_px(node);
+        top_left_radius_px.vertical_radius = top_left_radius.vertical_radius.resolved(node, height_length).to_px(node);
+        top_right_radius_px.vertical_radius = top_right_radius.vertical_radius.resolved(node, height_length).to_px(node);
+    } else /*(relative_to_width_only == RelativeToWidthOnly::Yes) */ {
```

Comment:
Use a switch case if you don't think if/else is clear enough without this comment